sklearn/metrics/_classification.py

```diff
@@ -2997,15 +2997,23 @@ def hamming_loss(y_true, y_pred, *, sample_weight=None):
     y_type, y_true, y_pred = _check_targets(y_true, y_pred)
     check_consistent_length(y_true, y_pred, sample_weight)
 
+    xp, _, device = get_namespace_and_device(y_true, y_pred, sample_weight)
+
     if sample_weight is None:
         weight_average = 1.0
     else:
-        weight_average = np.mean(sample_weight)
+        sample_weight = xp.asarray(sample_weight, device=device)
+        weight_average = _average(sample_weight, xp=xp)
 
     if y_type.startswith("multilabel"):
-        n_differences = count_nonzero(y_true - y_pred, sample_weight=sample_weight)
-        return float(
-            n_differences / (y_true.shape[0] * y_true.shape[1] * weight_average)
+        if _is_numpy_namespace(xp):
+            n_differences = count_nonzero(y_true - y_pred, sample_weight=sample_weight)
+        else:
+            n_differences = _count_nonzero(
+                y_true - y_pred, xp=xp, device=device, sample_weight=sample_weight
+            )
```

Comment:
I think we can remove this branching as _count_nonzero checks for the numpy namespace.
https://github.com/scikit-learn/scikit-learn/blob/035d65096b153edf07c3e2791e64755d83d33f2d/sklearn/utils/_array_api.py#L1066-L1067
```suggestion
        n_differences = _count_nonzero(
            y_true - y_pred, xp=xp, device=device, sample_weight=sample_weight
        )
```